src/main/java/com/hartwig/platinum/kubernetes/KubernetesClientProxy.java

```diff
@@ -36,15 +37,29 @@ public KubernetesClientProxy(String clusterName, GcpConfiguration gcpConfigurati
     }
 
     public MixedOperation<ConfigMap, ConfigMapList, Resource<ConfigMap>> configMaps() {
-        return kubernetesClient.configMaps();
+        return executeWithRetries(() -> kubernetesClient.configMaps());
     }
 
     public MixedOperation<Secret, SecretList, Resource<Secret>> secrets() {
-        return kubernetesClient.secrets();
+        return executeWithRetries(() -> kubernetesClient.secrets());
     }
 
     public NonNamespaceOperation<Job, JobList, ScalableResource<Job>> jobs() {
-        return kubernetesClient.batch().jobs().inNamespace(KubernetesCluster.NAMESPACE);
+        try {
+            return kubernetesClient.batch().jobs().inNamespace(KubernetesCluster.NAMESPACE);
+        } catch (KubernetesClientException e) {
+            authorise();
+            return jobs();
+        }
+    }
+
+    private MixedOperation executeWithRetries(Provider<MixedOperation<?, ?, ?>> operationProvider) {
+        try {
+            return operationProvider.get();
+        } catch (KubernetesClientException e) {
+            authorise();
+            return executeWithRetries(operationProvider);
```

Comment:
is the infinite recursion intentional? Feels like if the first authorise doesn't fix the issue the next one won't

Reply:
It is intentional just as the simplest implementation. If we start seeing it doing multiple reauths then I suppose we'd want to do something different.